core/src/main/java/io/questdb/griffin/model/RuntimeIntervalModel.java

```diff
@@ -162,19 +162,22 @@ private void addEvaluateDynamicIntervals(LongList outIntervals, SqlExecutionCont
 
                     outIntervals.extendAndSet(divider + 1, hi);
                     outIntervals.setQuick(divider, lo);
+                    if (divider == 0 && negated) {
```

Comment:
I'm not sure why `divider==0` condition is here, could you explain please?

Reply:
When `divider > 0` and the operation is subtraction, there will be a later `IntervalUtils.subtract(outIntervals, divider)` call on L192. But if the dynamic interval is the very first one, we need to negate it here.

Reply:
I think I now got why, nw